### PR TITLE
feat: Use v1 registry for tags from `quay.io`

### PR DESCRIPTION
**Pull request by eyalzek**
Since on v2 it returns unordered tags which breaks in certain cases

## Changes:

Switch to using v1 registry when fetching tags from `quay.io`.

## Context:

Fetching tags from quay using v2 returns tags in an unordered fashion. This means that in large repositories it's possible to never actually get the "latest" tag available. Using v1 in case the registry is quay should mitigate this issue. Relevant issue https://github.com/renovatebot/renovate/issues/9176

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/docker/index.ts [hunk 1]
```diff
@@ -448,19 +448,32 @@ async function getTags(
     // AWS ECR limits the maximum number of results to 1000
     // See https://docs.aws.amazon.com/AmazonECR/latest/APIReference/API_DescribeRepositories.html#ECR-DescribeRepositories-request-maxResults
     const limit = ecrRegex.test(registry) ? 1000 : 10000;
-    let url = `${registry}/v2/${repository}/tags/list?n=${limit}`;
+
+    let url: string;
+    let page = 1;
+
+    // quay registry returns unordered tags using v2, let's use v1 for now
+    const isQuay = registry.endsWith('.quay.io');
```
Comment (rarkins) [inline lib/datasource/docker/index.ts:456]: Isn't the registry usually without subdomain, e.g. `quay.io` or `https://quay.io`?

File: lib/datasource/docker/index.ts [hunk 2]
```diff
@@ -448,19 +448,32 @@ async function getTags(
...
-    let page = 1;
     do {
+      page += 1;
       const res = await http.getJson<{ tags: string[] }>(url, { headers });
       tags = tags.concat(res.body.tags);
-      const linkHeader = parseLinkHeader(res.headers.link as string);
-      url = linkHeader?.next ? URL.resolve(url, linkHeader.next.url) : null;
-      page += 1;
+      if (isQuay) {
+        url = res.body.tags.length > 0 ? `${registry}/api/v1/repository/${repository}/tag/?limit=${limit}&page=${page}` : null;
```
Comment (rarkins) [inline lib/datasource/docker/index.ts:472]: Does this definitely handle all cases? e.g. are you sure that we can just keep "blindly" paginating until one day it returns zero results?

File: lib/datasource/docker/index.ts (hunk 1 above)
Comment (eyalzek) [reply lib/datasource/docker/index.ts:456]: Yes, good point.

File: lib/datasource/docker/index.ts (hunk 2 above)
Comment (eyalzek) [reply lib/datasource/docker/index.ts:472]: No, I'm not sure of that at all. Tested only using curl with the repo from the issue. This requires further testing.

**Pull request closed without merging**